File: IT/2/Interfacce grafiche/Esercizi/C++/esercizio17/esercizio17.cpp
/* file esercizio17.cpp */

#include <QApplication>
#include <QVBoxLayout>
#include <QMenuBar>
#include "esercizio17.h"

Esercizio17::Esercizio17(QWidget *parent) : QMainWindow(parent)
{
	QMenu *fileMenu = menuBar()->addMenu("&File");

	QAction *closeAction = new QAction("Esci", this);
	fileMenu->addAction(closeAction);

	connect(closeAction, SIGNAL(triggered()), this, SLOT(closeHandler()));

	text = new QTextEdit();
	button1 = new QPushButton("Testa", this);
	connect(button1, SIGNAL(clicked()), this, SLOT(btn1ClickHandler()));
	button1->show();

	button2 = new QPushButton("Croce", this);
	connect(button2, SIGNAL(clicked()), this, SLOT(btn2ClickHandler()));
	button2->show();

	QWidget *widget = new QWidget(this);

	QVBoxLayout *layout = new QVBoxLayout(widget);
	layout->addWidget(text);
	layout->addWidget(button1);
	layout->addWidget(button2);
	widget->setLayout(layout);

	this->setCentralWidget(widget);

	this->resize(280, 250);
	this->setWindowTitle("Testa o croce?");
	this->show();
}

void Esercizio17::btn1ClickHandler()
{
	text->append("Testa");
}

void Esercizio17::btn2ClickHandler()
{
	text->append("Croce");
}

void Esercizio17::closeHandler()
{
	QApplication::quit();
}

File: IT/2/Interfacce grafiche/Esercizi/Python/esercizio17.py
# file esercizio17.py

try:
	from tkinter import Frame, Button, Menu, INSERT
	from tkinter.scrolledtext import ScrolledText
except ImportError:
	from Tkinter import Frame, Button, Menu, INSERT
	from ScrolledText import ScrolledText

class Widget(Frame):
	def __init__(self, parent):
		Frame.__init__(self, master=parent, width=280, height=170)
		self.scrolledText = ScrolledText(self)
		self.scrolledText.pack()
		self.pack_propagate(False)
		self.pack()

class Esercizio17(Frame):
	def __init__(self):
		Frame.__init__(self, master=None)

		self.menubar = Menu(self)
		fileMenu = Menu(self.menubar)
		fileMenu.add_command(label="Esci", command=self.closeHandler)
		self.menubar.add_cascade(label="File", menu=fileMenu)

		self.widget = Widget(self)
		self.button1 = Button(self, text="Testa", command=self.btn1ClickHandler)
		self.button1.pack()
		self.button2 = Button(self, text="Croce", command=self.btn2ClickHandler)
		self.button2.pack()

		self.master.config(menu=self.menubar)
		self.pack()
		self.master.wm_title("Testa o croce?")
		self.mainloop()

	def btn1ClickHandler(self):
		self.widget.scrolledText.insert(INSERT, "Testa\n")

	def btn2ClickHandler(self):
		self.widget.scrolledText.insert(INSERT, "Croce\n")

	def closeHandler(self):
		self.master.destroy()

Esercizio17()

